src/cbrkit/sim/graphs/alignment.py

```diff
@@ -0,0 +1,168 @@
+        >>> nodes_y = Map({
+        ...     '1': Node(key='1', value='A'),
+        ...     '2': Node(key='2', value='C'),
+        ... })
+        >>> edges_y = Map({
+        ...     'e1': Edge(key='e1', source=nodes_y['1'], target=nodes_y['2'], value='Y'),
+        ... })
+        >>> graph_y = Graph(nodes=nodes_y, edges=edges_y, value=None)
+        >>> g_dtw = dtw(node_similarity, edge_similarity)
+        >>> result = g_dtw(graph_x, graph_y)
+        >>> print(result)
+        GraphSim(value=0.05555555555555555, node_mapping={'1': '1', '2': '2'}, edge_mapping={'e1': 'e1'})
+
+        >>> # Example with BatchSimFunc
+        >>> def batch_node_similarity(pairs):
+        ...     return [1.0 if n1.value == n2.value else 0.0 for n1, n2 in pairs]
+        >>> def batch_edge_similarity(pairs):
+        ...     return [1.0 if e1.value == e2.value else 0.0 for e1, e2 in pairs]
+        >>> g_dtw_batch = dtw(batch_node_similarity, batch_edge_similarity)
+        >>> result_batch = g_dtw_batch(graph_x, graph_y)
+        >>> print(result_batch)
+        GraphSim(value=0.05555555555555555, node_mapping={'1': '1', '2': '2'}, edge_mapping={'e1': 'e1'})
+    """
+
+    node_sim_func: AnySimFunc
+    edge_sim_func: AnySimFunc | None = None
+    normalize: bool = True
+
+    def __call__(
+        self,
+        x: Graph[K, N, E, G],
+        y: Graph[K, N, E, G],
+    ) -> GraphSim[K]:
+        # Use the updated to_sequence function
+        sequence_x, edges_x = to_sequence(x)  # sequence_x: list[N], edges_x: list[E]
+        sequence_y, edges_y = to_sequence(y)  # sequence_y: list[N], edges_y: list[E]
+
+        # Wrap BatchSimFunc if necessary
+        wrapped_node_sim_func = self._wrap_batch_func(self.node_sim_func)
+        wrapped_edge_sim_func = (
+            self._wrap_batch_func(self.edge_sim_func)
+            if self.edge_sim_func is not None
+            else None
+        )
+
+        # Use the dtwmodule for node distances
+        node_result = dtwmodule(distance_func=wrapped_node_sim_func)(
+            sequence_x, sequence_y, return_alignment=True
+        )
+
+        # Extract alignment and node distance
+        alignment = node_result.mapping
+        node_distance = node_result.value
+        node_similarity = self._compute_similarity_from_func(
+            self.node_sim_func, alignment
+        )
+
+        node_mapping = {
+            y_node.key: x_node.key for x_node, y_node in alignment if x_node and y_node
+        }
+
+        edge_similarity = None
+        edge_mapping = {}
+
+        if self.edge_sim_func:
+            # Use the dtwmodule for edge distances
+            edge_result = dtwmodule(distance_func=wrapped_edge_sim_func)(
+                edges_x, edges_y, return_alignment=False
+            )
+
+            edge_distance = edge_result.value
+            edge_similarity = self._compute_similarity_from_func(
+                self.edge_sim_func, zip(edges_x, edges_y)
+            )
```

Comment:
Similar to the mentioned `unbatchify_sim`, there is also a `batchify_sim` helpers that could be used here.